### PR TITLE
Use the correct library when a relative path is passed as `--library` value

### PR DESCRIPTION
**Pull request by MatteoPologruto**
## Please check if the PR fulfills these requirements

See [how to contribute](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/)

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)
- [ ] `configuration.schema.json` updated if new parameters are added.

## What kind of change does this PR introduce?
Code imperfection fix

## What is the current behavior?
Compiling an example of a manually downloaded library fails if  the command is executed inside the examples directory and a relative path to the library itself is specified.
```
C:\Users\m.pologruto>arduino-cli lib uninstall ArduinoIoTCloud
Uninstalling [ArduinoIoTCloud@1.11.0]...

C:\Users\m.pologruto>cd Downloads/ArduinoIoTCloud-1.11.0/examples/ArduinoIoTCloud-Basic

C:\Users\m.pologruto\Downloads\ArduinoIoTCloud-1.11.0\examples\ArduinoIoTCloud-Basic>arduino-cli compile -b arduino:samd:nano_33_iot --library ../../
...
...
Error during build: CreateFile ..\..\..\src\ArduinoIoTCloud.cpp: The system cannot find the path specified.
```
If the same library is installed with `lib install`, the library is picked from the Sketchbook's libraries directory.

## What is the new behavior?
If the relative path to a library is correct, that library will be chosen over other instances of it.

## Does this PR introduce a breaking change, and is [titled accordingly](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#breaking)?
No

### CODE REVIEW

**Review by per1234 — CHANGES_REQUESTED** [review]
There is no need to bundle the entire ArduinoIoTCloud library as test data. You can use a minimal library instead:

[FooLib.zip](https://github.com/arduino/arduino-cli/files/11085270/FooLib.zip)

The only requirement for the test data:

- A sketch that contains an `#include` directive.
- A library that:
  - contains a header file with name matching the `#include` directive in the sketch
  - contains at least one source file (e.g., `FooLib.cpp`)

File: internal/integrationtest/compile_3/compile_test.go [hunk 1]
```diff
@@ -116,3 +116,48 @@ func TestCompilerErrOutput(t *testing.T) {
...
+	// Install library and its dependencies
+	zipPath, err := paths.New("..", "testdata", "FooLib.zip").Abs()
+	require.NoError(t, err)
+	_, _, err = cli.Run("lib", "install", "--zip-path", zipPath.String(), "--config-file", configFile.String())
+	require.NoError(t, err)
+
+	// Uninstall library
+	_, _, err = cli.Run("lib", "uninstall", "FooLib")
+	require.NoError(t, err)
+
```
Comment (per1234) [inline internal/integrationtest/compile_3/compile_test.go:143]:
```suggestion
```

This code doesn't serve any purpose that I can see

**Review by per1234 — CHANGES_REQUESTED** [review]

File: commands/compile/compile.go [hunk 2]
```diff
@@ -128,7 +128,12 @@ func Compile(ctx context.Context, req *rpc.CompileRequest, outStream, errStream
 	// FIXME: This will be redundant when arduino-builder will be part of the cli
 	builderCtx.OtherLibrariesDirs = paths.NewPathList(req.GetLibraries()...)
 	builderCtx.OtherLibrariesDirs.Add(configuration.LibrariesDir(configuration.Settings))
-	builderCtx.LibraryDirs = paths.NewPathList(req.Library...)
+	for _, libPath := range paths.NewPathList(req.Library...) {
+		if libPath, err = libPath.Abs(); err != nil {
+			return nil, &arduino.LibraryNotFoundError{Cause: err}
+		}
+		builderCtx.LibraryDirs = append(builderCtx.LibraryDirs, libPath)
+	}
```
Comment (cmaglie) [inline commands/compile/compile.go:136]: Move this change from `commands` package (gRPC), to `internal/cli` package (arduino-cli).

It doesn't make much sense to use relative paths in a gRPC call as a daemon. It makes sense in the `arduino-cli` as a command line, so the paths must be absolutized before entering the gRPC request.

**Review by cmaglie — CHANGES_REQUESTED** [review]

**Review by per1234 — APPROVED** [review]
Thanks Matteo!

**Review by cmaglie — APPROVED** [review]

**Pull request merged**